testing/test-helper.js

```diff
@@ -45,6 +45,27 @@ export function stubServiceForDoc(sandbox, ampdoc, serviceId, method) {
 }
 
 /**
+ * Resolves a promise when the method stub is called for the first time.
+ */
+export function stubWithCalledPromise(sandbox, object, method, opt_callMethod) {
+  const original = object[method];
+  let stub;
+  const promise = new Promise(resolve => {
+    stub = sandbox.stub();
+    const spy = sandbox.stub(object, method, () => {
+      const result = stub.apply(object, spy.lastCall.args);
```

Comment:
The short answer is because the Sinon stub API is weird.

The long answer is Sinon doesn't implement a promise API for when a stub is called, so I'm implementing that here. The stub constructor does accept a callback argument, which I use to resolve the promise, but it returns a spy instance instead of a stub instance. If a test needs to change a stub's behavior, e.g. calling `stub.returns(...)`, it has no reference to the stub. Even `object.method` is a spy if you pass a callback argument to `sandbox.stub`.

We could avoid using a spy and a stub together using `Stub#callsFake` in Sinon 2.0, but we are using 1.17